libs/react-helpers/src/hooks/use-mutation-observer.ts

```diff
@@ -0,0 +1,45 @@
+import { captureException } from '@sentry/react';
+import debounce from 'lodash/debounce';
+import { useEffect, useState } from 'react';
+
+type MutationObserverConfiguration = {
+  debounceTime: number;
+  config: MutationObserverInit;
+};
+
+const DEFAULT_OPTIONS: MutationObserverConfiguration = {
+  debounceTime: 1000 / 16,
+  config: {
+    attributes: true,
+    childList: false,
+    subtree: false,
+  },
+};
+
+export function useMutationObserver(
+  target: Node | null,
+  callback: MutationCallback,
+  options: MutationObserverConfiguration = DEFAULT_OPTIONS
+) {
+  const [observer, setObserver] = useState<MutationObserver | null>(null);
+
+  useEffect(() => {
```

Comment:
suggestion (non blocking): I'm not a fan of setting state inside useEffects that then trigger other useEffects, it makes the code less linear and hard to read. I think the below should suffice? 

```
useEffect(() => {
  if (!target) return
  
  const observer = new MutationObserver(...)
  
  try {
    ...
  } catch() {
    ...
  } 
  
  return () => observer.disconnect()
}, [options, target, callback])
```

Reply:
Changed to `useMemo`